### ISSUE TITLE
Series Management

### ISSUE THREAD

**Issue by NCrusher74** [TortugaPower/BookPlayer#428]
I don't know how I didn't stumble upon this app last spring when I first started listening to AudioBooks, but I really love it so far.

There are just a couple features that I can't find ANYWHERE, and I've even started learning Swift in order to make my own apps that include them (particularly a library maintenance app for non-DRMed audiobooks for MacOS) but while I'm steadily plugging away at that, I'm not at a level yet where I'd be capable of messing with someone else's working code without needing a lot of guidance and hand-holding. I'd be happy to try to implement these features, but I don't have the skills to do it on my own.

Forgive me for the excess verbiage, but it's hard to make this recommendation without a lot of wordy context.

Basically, as someone who reads/listens to a LOT of series, some of which are series within the chronology of a larger universe (for example, think of Star Wars as a "Universe" containing three, three-book series.)

This can make getting files to fall into the right order difficult without a lot of wordy and/or misleading file-naming and/or metadata tagging. For instance, this is an example of what The Hobbit and The Lord of the Rings look like on my drive:

<img width="601" alt="Screen Shot 2020-02-14 at 3 33 46 PM" src="https://user-images.githubusercontent.com/56804260/74576626-81a81900-4f40-11ea-9dae-a7b515261f0c.png">

and this is what the metadata looks like in Subler:

<img width="771" alt="Screen Shot 2020-02-14 at 3 34 23 PM" src="https://user-images.githubusercontent.com/56804260/74576771-2cb8d280-4f41-11ea-9e5b-9f5f3c3dc2ad.png">

Even with some rather long file-names, the Foreword and Prologue of "The Fellowship of the Ring" still get sorted incorrectly. Some of this can be fixed using track numbers, but only if you don't have a book that is in multiple .mp3 or .m4a files (I divided up the parts of the LOTR books to demonstrate this.)

My thought was to use a metadata field that is generally unused (or a custom one) for "Universe", "Series" and their indices. (demonstrated here using TV Show and Episode and Season tags)

<img width="297" alt="Screen Shot 2020-02-14 at 3 37 18 PM" src="https://user-images.githubusercontent.com/56804260/74576860-9c2ec200-4f41-11ea-8a13-a59aad418799.png">

For example "The Hobbit" might be (universe) "Middle Earth" Book 1 (assuming you don't have  The Simarillion and so forth in your library), but it (technically) wouldn't be part of (series) "The Lord of the Rings".

"The Fellowship of the Ring", however, would be Middle Earth, Book 2, but Lord of the Rings, Book 1. Or maybe "The Lord of the Rings" would be Middle Earth, Book 2, and you'd slot the series in that way, but that only works if you don't have books in a universe that don't interleave between series (see also: David Weber's "Honor Harrington"/"Wages of Sin"/"Saganami Island" series, not to mention the short story anthologies.) If read in chronological order, those fall as follows:

```
-     Honor Among Enemies (Honor Harrington #6)
-     "A Whiff of Grapeshot" (short story in More than Honor, Worlds of Honor Anthology #1)
-     "Changer of Worlds" (short story in Changer of Worlds, Worlds of Honor Anthology #3)
-     In Enemy Hands (Honor Harrington #7)
-     Echoes of Honor (Honor Harrington #8)
-     Ashes of Victory (Honor Harrington #9)
-     "From the Highlands" (Short story in Changer of Worlds)
-     "Nightfall" (Short story in Changer of Worlds)
-     "Fanatic" (Short story in The Service of the Sword, Worlds of Honor Anthology #4)
-     War of Honor (Honor Harrington #10)
-     Crown of Slaves (Wages of Sin #1 - for which "From the Highlands" and "Fanatic" provide necessary background)
-     The Shadow of Saganami (Saganami Island #1)
-     At All Costs (Honor Harrington #11)
-     Storm from the Shadows (Saganami Island #2)
-     Torch of Freedom (Wages of Sin #2)
-     Mission of Honor (Honor Harrington #12)
-     A Rising Thunder (Honor Harrington #13)
-     Shadow of Freedom (Saganami Island #3)
-     Cauldron of Ghosts (Wages of Sin #3)
-     Shadow of Victory (Saganami Island #4)
-     Uncompromising Honor (Honor Harrington #14)
```

So anyway, my recommendation is for the app to be able to read a certain metadata field for this information, and construct a playlist (which might actually be called a "Series" list to differentiate it from a straight-up playlist) based on how the user wants the books ordered.

And then to display the series title on a separate line from the book title, so that the title field isnt so long:

<img width="781" alt="BP Meta with Series and Chapter Nav" src="https://user-images.githubusercontent.com/56804260/74577168-bae18880-4f42-11ea-89ef-633b4f7f38f5.png">

(also, Chapter Nav buttons should totally be a thing.)

<img width="769" alt="Screen Shot 2020-02-14 at 3 59 47 PM" src="https://user-images.githubusercontent.com/56804260/74577266-2c213b80-4f43-11ea-8eff-f9109441e6de.png">

I don't know if Apple CarPlay allows for the metadata fields to be more than the three lines that are there, but something like this would be awesome, too.

<img width="524" alt="Screen Shot 2020-02-14 at 4 00 25 PM" src="https://user-images.githubusercontent.com/56804260/74577259-23c90080-4f43-11ea-86a6-807e73e3a5c6.png">

Thank you!

**Comment by pichfl** [comment]
No need to apologize for a long request, it might just be an indicator, that you're actually requesting multiple things. I'll try to answer as much as I can right away. I also extracted #429 to handle the discussion for the chapter navigation.

**A quick solution for your problem from ancient days:** Start your file names with a zero prefixed number. Sorting is applied alphanumerically, which means numbers from `001` to `999` will be sorted reliably before any word characters. If metadata is filed correctly, the file name is used for sorting only and will not show up in the UI otherwise.

**Feature discussion**
Given I listen to many longer series myself, I like the idea of having series. 

However, using custom metadata as the only way to create this, is not something we can rely on. 

While your audiobook collection looks meticulously maintained (consider me impressed), we have many users who just want to dump a folder of MP3 files onto BookPlayer and we have to make sure it works for them as well.

Also, there are ideas to create bound books from single files (see #389), which could provide a way to handle more complex books. We haven't discussed how binding should handle files which include chapter metadata yet, which we should and I just added a comment there as well.

There is also #260 which discusses having a screen for displaying and *editing* book metadata in a secondary screen to the player, which will be realized with the upcoming card based player redesign (#281 and some other work which is not ready for the public yet)

Both #260 and #389 these need to be handled before starting this issue so everyone gets a chance to use it.

We also need to make sure we do not introduce too many layers of abstraction and organization into the UI so the app stays easy to use. 

Our goal by name is to provide a good listening experience, not to manage your audiobook collection. We try to provide very easy ways to import files at any time (and macOS and Airdrop help a lot) so you don't have have all your books on your phone all the time but just carry a few you're currently listening to. (Which corresponds with the fact that many users will continue to have phones that will not fit larger collections anyway)

**Comment by NCrusher74** [comment]
> While your audiobook collection looks meticulously maintained (consider me impressed), we have many users who just want to dump a folder of MP3 files onto BookPlayer and we have to make sure it works for them as well.

True. My rationale for beginning to learn Swift and making my project an audiobook library manager is the fact that I'm a moderately obsessive bibliophile who surrounds herself with other bibliophiles, I tend to forget that not everyone is that obsessive about their libraries (jeez, Nolaine, relax, some people just wanna chill with a book and don't care, okay?)

Series management has always been a bugaboo of mine (even Calibre, peerless as it is, doesn't quite manage it the way I prefer for my ebooks, but it's at least flexible enough that I can work with it.) But I totally get that that isn't what BookPlayer is geared toward.

Thank you for taking the time to respond. I look forward to seeing where BookPlayer goes as it continues to develop. Assuming I can ever figure out how to write metadata instead of just reading it, maybe I can even help contribute someday when you work on that metadata editing feature.

**Comment by pichfl** [comment]
I don't want to keep you from learning Swift or contributing though! We have many issues marked as [help wanted](https://github.com/TortugaPower/BookPlayer/labels/help%20wanted) were you can contribute. I also marked a few of them as [good for beginners](https://github.com/TortugaPower/BookPlayer/labels/good%20for%20beginners), which means they should not require super complex code changes.

**Comment by steve4jobs4** [comment]
It would be nice to have an optional cover view. For instance in each row two or three audiobook artworks / covers. But only the artworks without any text below. And that you can switch between cover view and the regular list view.

![Konzept](https://user-images.githubusercontent.com/62400793/77110295-a69f1880-6a25-11ea-8ca6-ecdcf596176c.jpeg)

The second idea that I have in mind is an option which allows you to completly set the player controlls to 0% opacity. And that they reappear when you click on the artwork. At the moment they don‘t become completely transparent. 

![IMG_7221](https://user-images.githubusercontent.com/62400793/77110590-16ad9e80-6a26-11ea-8028-80e9d7120386.jpeg)


And I would love to have an ipad version of the app.

Thank you for the amazing app. I hope you are considering my ideas.

**Comment by pichfl** [comment]
@steve4jobs4 I extracted all your requests into separate issues, as they are not quite related to this one here. Since you're new to GitHub: Please be mindful about other peoples time, issues should be focussed on a single topic. 

- See #437 for the enhancement request for a library view based on cover images
- See #438 for discussion on the controls
- See #230 (which has been around for more than a year now) for an iPad version

**Comment by pichfl** [comment]
I'm closing this issue for now as the feature is most likely to complex for BookPlayer in its current iteration. 

Off-Topic: @NCrusher74 I hope to see you in our Discord at some point. A Swift based organizer for audiobooks that interfaces with BookPlayer is something I'd very interested in.
